src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,30 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) object;
+        return width != null && width.equals(rectangle.width) || rectangle.width == width
+                && length != null && length.equals(rectangle.length) || rectangle.length == length
+                && color != null && color.equals(rectangle.color) || rectangle.color == color;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 17;
+        int modifier1 = width == null ? 1 : width;
+        int modifier2 = length == null ? 1 : length;
+        int modifier3 = color == null ? 1 : color.hashCode();
+        result = 31 * result * modifier1;
```

Comment:
```suggestion
        result = 31 * result * width == null ? 1 : width.hashCode();
```
please refactor all three lines like this - remove `modifier` variables and use hashCode of the fields